### PR TITLE
NET-sthinbetween-20201203

### PR DESCRIPTION
**Pull request by sthinbetween**
Muss aus den Folien noch ein wenig nachgetragen werden, sobald diese verfügbar sind. 
Zudem fehlt noch die Doku des Praxisbeispiels

### CODE REVIEW

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Review by Steve2955 — CHANGES_REQUESTED** [review]
@sthinbetween Ich bin bei der Vorlesung zum Schluss auch kaum hinterhergekommen, ich würde warten bis er die Präsentation zur Verfügung gestellt, dann kann ich gerne wieder alles mergen und den Rest noch ergänzen

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by sthinbetween** [comment]
So, der heutige Kram ist auch drin.

File: NET/README.md [hunk 1]
```diff
@@ -955,3 +991,476 @@ Flags: Bitflags zur Steuerung der Kommunikation (z.B. Aufbau, Trennung, der Verb
...
+
+Begriffsunterscheidung: 
+- Forwarding Information Base (FIB) und Routingtabellen unterscheiden sich eigentlich, sind jedoch in Theorie und Praxis nicht einheitlich bezeichnet 
+
+
+### Schema für IP-Forwarding Algorithmus
+
+```
+Ziel: D 
+N = Präfix nzw. Netz von D
```
Comment (RvNovae) [inline NET/README.md:1041]:
```suggestion
N = Präfix bzw. Netz von D 
```

File: NET/README.md [hunk 2]
```diff
@@ -955,3 +991,476 @@ Flags: Bitflags zur Steuerung der Kommunikation (z.B. Aufbau, Trennung, der Verb
...
+  - RIPng (mit IPv6 Unterstützung)
+
+<!-- TODO: Rest von Folien übernehmen  -->
+
+Schwächen: 
+- Hop-Limit von 15
+  - Falls ein Knoten über mehr als 15 Hops erreichbar ist, gilt er für RIP als unerreichbar
+- keine Separation in Broadcast-Domänen 
+  - Netz wird durch Broadcasts geflutet
+  - keine Möglichkeit, große Netze in RIP:Bereiche einzuteilen, die nur innerhlab des Bereichs Informationen austauschen
```
Comment (RvNovae) [inline NET/README.md:1189]:
```suggestion
  - keine Möglichkeit, große Netze in RIP:Bereiche einzuteilen, die nur innerhalb des Bereichs Informationen austauschen
```

File: NET/README.md [hunk 3]
```diff
@@ -955,3 +991,476 @@ Flags: Bitflags zur Steuerung der Kommunikation (z.B. Aufbau, Trennung, der Verb
...
+  <!-- TODO: Zitat aus Folie 21 übernehmen  -->
+
+![BGP-Nutzungsmotivation Überblick](resources/routing-as-bgp.png)<!-- width=500px -->
+
+- zum Beispiel: 
+  - sobald eine Entscheidungsmöglichkeit mehrerer Routen (über mehrere ISP) vorliegt, lohnt sich der Einsatz von BGP 
+
+#### BGP-Peers /-Nachbarschaften
+
+- BGP- Router etabliert Verbindung mit ausgewählten Peers und tauscht mit diesen Pfadinformationen aus
```
Comment (RvNovae) [inline NET/README.md:1265]:
```suggestion
- BGP-Router etabliert Verbindung mit ausgewählten Peers und tauscht mit diesen Pfadinformationen aus 
```

File: NET/README.md [hunk 4]
```diff
@@ -955,3 +991,476 @@ Flags: Bitflags zur Steuerung der Kommunikation (z.B. Aufbau, Trennung, der Verb
...
+#### BGP-Pfadauswahl 
+
+- RFC 4271 macht keine Vorgaben zur Selektion eines Pfades 
+- Informationen zur Selektion sind sehr allgemein: 
+  - 
+  - 
+- Möglicher Ansatz im folgenden Beispiel: Wahl der Route mit dem kürzesten AS-Pfad (via AS 500)
+
+
+
```
Comment (RvNovae) [inline NET/README.md:1305]: fehlt hier was? 🤔

File: NET/README.md [hunk 5]
```diff
@@ -955,3 +991,476 @@ Flags: Bitflags zur Steuerung der Kommunikation (z.B. Aufbau, Trennung, der Verb
...
+  - Fehlkonfigurationen der AS können schwerwiegende Folgen haben 
+  - Beispiel malaysischer Provider mit AS-Beziehungen zu Nachbarn 
+    - Fehlkonfiguration: 
+      - Ziel-IP-Adressen im gesamten Bereich wurden mit Kosten von 0 definiert
+      - Information wurde von vielen AS verteilt
+      - daher wurden massiver Traffic über diesen einen Provider geschleust 
+      - -> große Teile des Internet zusammengebrochen 
+    - Problem hier: 
+      - keine Validation der propagierten Informationen 
+  - Beispiel Hyjack eines BGP-Routers
```
Comment (RvNovae) [inline NET/README.md:1324]:
```suggestion
  - Beispiel Hijack eines BGP-Routers
```

File: NET/README.md [hunk 6]
```diff
@@ -955,3 +991,476 @@ Flags: Bitflags zur Steuerung der Kommunikation (z.B. Aufbau, Trennung, der Verb
...
+  - Sitzungsschicht
+  - Darstellungsschicht
+  - Anwendungsschicht
+
+- die Anwendungsschicht setzt dabei auf der Transportschicht auf und greift auf diese beispielsweise über die Socket-Schnittstelle zu
+
+- Unterscheidung von zwei Formen von Anwendungsprotokollen gemäß RFC 1122
+  - Anwenderprotokolle: 
+    - HTTP, SMTP, SSH 
+  - Unterstü
```
Comment (RvNovae) [inline NET/README.md:1354]:
```suggestion
  - Unterstützende Protokolle
```
Eventuell gibts da auch nen richtigen Begriff, hab aber auf die schnelle keinen gefunden.

File: NET/README.md [hunk 7]
```diff
@@ -955,3 +991,476 @@ Flags: Bitflags zur Steuerung der Kommunikation (z.B. Aufbau, Trennung, der Verb
...
+    - in der config stehen die Nameserver z.B. des ISP 
+  - bisher hat der Nameserver des ISP aber die Information über die neue Domain noch gar nicht
+    - Trennung der Domain in die TLD und den TLD-spezifischen Namen
+    - ISP-Nameserver steuert daraufhin einen der 13 Root-NS an 
+    - dort wird angefragt, wer für die TLD verantwortlich ist 
+    - Root-Server gibt `resource-record` für den TLD-Nameserver zurück 
+  - für .de -> DENIC (eingetragener Verein)
+    - diese betreiben eine ganze Reihe von NS
+  - daraufhin Anfrage an NS von DENIC
+    - Wer ist für den Namen der Domain verantwortlich? (hier: br1it -> Strato)
```
Comment (RvNovae) [inline NET/README.md:1391]: Die Beispieldomain wurde vorher nie erwähnt, könnte eventuell etwas verwirren. Ansonsten aber nicht schlimm, wollts nur gesagt haben.

File: NET/README.md [hunk 8]
```diff
@@ -955,3 +991,476 @@ Flags: Bitflags zur Steuerung der Kommunikation (z.B. Aufbau, Trennung, der Verb
...
+    - DNS gibt `resource-record` für Strato-NS zurück 
+    - Eintrag von Strato auf dem DENIC-NS erfolgte bei Registrierung der Domain
+  - daraufhin Anfrage bei Strato-NS 
+    - dieser gibt dann `resource-record` für die Domain zurück 
+  - NS des ISP leitet diesen an Resolver des Nutzers weiter 
+    - Dieser hat dann die IP des Webservers und kann auf diesen zugreifen 
+  - Informationen werden auf allen Ebenen teilweise gecacht:
```
Comment (RvNovae) [inline NET/README.md:1398]:
```suggestion
  - Informationen werden auf allen Ebenen teilweise gecached: 
```

**Review by RvNovae — CHANGES_REQUESTED** [review]
Ein paar kleinere Anmerkungen (größtenteils typos). 

Ansonsten sehr ausführlich, vielen Dank!

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

File: NET/README.md (hunk 4 above)
Comment (sthinbetween) [reply NET/README.md]: fehlte, ja ;)

File: NET/README.md (hunk 7 above)
Comment (sthinbetween) [reply NET/README.md]: Done

**Comment by Steve2955** [comment]
## Command results
<details>
<summary>Details: </summary>

<details>
<summary><em>Run doctoc</em></summary>

```Shell
changed:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/NET/README.md
unchanged:
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/DBS/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/RES/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/PRO/README.md
  - /home/runner/work/dhge-pi19-sem3/dhge-pi19-sem3/SWE/README.md
```



</details>

</details>

## Changed files
<details>
<summary>Changed file: </summary>

- NET/README.md

</details>

<hr>

[:octocat: Repo](https://github.com/technote-space/toc-generator) | [:memo: Issues](https://github.com/technote-space/toc-generator/issues) | [:department_store: Marketplace](https://github.com/marketplace/actions/toc-generator)

**Comment by sthinbetween** [comment]
Schaut gern nochmal durch, nachgetragen ist jetzt alles. 
Was noch fehlt, aber gern in einem nächsten Branch gemacht werden kann, ist die Beantwortung des ersten Probeklausurteils (Subnetzteilung habe ich mitgemacht)

**Review by ZeroPointMax — CHANGES_REQUESTED** [review]
es fehlen l2-net.png und transportschicht.png, sehe ich das richtig?

**Comment by sthinbetween** [comment]
> 
> 
> es fehlen l2-net.png und transportschicht.png, sehe ich das richtig?

@ZeroPointMax ich habe es auch gerade gesehen. Allerdings selbes Problem wie bei den anderen beiden Files: in meiner History ist das Löschen nicht erkennbar und im Main gibt es die auch nicht, obwohl sie da sein müssten.

**Comment by sthinbetween** [comment]
Hab den Übeltäter gefunden: im SWE PR vom 4.12. sind die betreffenden Files rausgeflogen. Warum auch immer

**Review by ZeroPointMax — APPROVED** [review]
Sieht jetzt gut aus, vielen Dank! 🚀

**Comment by ZeroPointMax** [comment]
@Steve2955 @RvNovae ihr seid als "requested changes"-Status geführt - Re-Review pls?

**Review by RvNovae — APPROVED** [review]

File: NET/README.md [hunk 9]
```diff
@@ -955,3 +1018,916 @@ Flags: Bitflags zur Steuerung der Kommunikation (z.B. Aufbau, Trennung, der Verb
...
+        falls distanz[a] + w < distanz[b]:
+          distanz[b] := distanz[a] + w
+          vorgaenger[b] := a
+    ``` 
+  - Abschließend wird geprüft, ob ein Zyklus mit negativem Gewicht existiert
+  - Nachteile für Routing-Algorithmen: 
+    - keine gute Skalierbarkeit 
+    - `Count-to-infinity-Problem`
+
+$/rightarrow$ Zahlreiche Erweiterungen bzw. Alternativen (insb. Dijkstra verfügbar)
```
Comment (Steve2955) [inline NET/README.md:1239]: Müsste das nicht ein `\rightarrow` sein?

Comment (sthinbetween) [reply NET/README.md:1239]: Stimmt. Das ist sicherlich nicht die einzige Stelle. Ich geh nach den VL heute nochmal drüber, hab leider meinen 2FA Token nicht dabei um das jetzt zu machen.

Comment (sthinbetween) [reply NET/README.md]: @Steve2955 Gesagt, getan.

**Review by Steve2955 — APPROVED** [review]

**Comment by RvNovae** [comment]
> @Steve2955 @RvNovae ist original niemandem aufgefallen, dass hier in Zeile 1796 "Dynamic **Horst** Configuration Protocol" steht 😂
> 
> Schuld hat @sthinbetween lol

das ist ja großartig

**Pull request merged**